File: actionlib_pkg/src/fibonacciActionClient.cpp
#include <ros/ros.h>
#include <actionlib/client/simple_action_client.h>
#include <actionlib/client/terminal_state.h>
#include <actionlib_pkg/fibonacciAction.h>

// http://wiki.ros.org/actionlib_tutorials/Tutorials/SimpleActionClient

int main (int argc, char **argv)
{
  ros::init(argc, argv, "fibonacciActionClient");
  ROS_INFO("fibonacciActionClient.cpp - Create node fibonacciActionClient");

  // create the action client
  // true causes the client to spin its own thread
  actionlib::SimpleActionClient<actionlib_pkg::fibonacciAction> ac("fibonacci", true);

  ROS_INFO("fibonacciActionClient.cpp - Waiting for action server to start.");
  // wait for the action server to start
  ac.waitForServer(); //will wait for infinite time

  ROS_INFO("fibonacciActionClient.cpp - Action server started, sending goal.");
  // send a goal to the action
  actionlib_pkg::fibonacciGoal goal;
  goal.order = 50;
  ac.sendGoal(goal);

  //wait for the action to return
  bool finished_before_timeout = ac.waitForResult(ros::Duration(goal.order + 5));

  if (finished_before_timeout)
  {
    actionlib::SimpleClientGoalState state = ac.getState();
    ROS_INFO("fibonacciActionClient.cpp - Action finished: %s",state.toString().c_str());
  }
  else
    ROS_INFO("fibonacciActionClient.cpp - Action did not finish before the time out.");

  //exit
  return 0;
}

File: actionlib_pkg/scripts/fibonacciActionClient.py
#! /usr/bin/env python

import rospy
# from __future__ import print_function
# Brings in the SimpleActionClient
import actionlib
# Brings in the messages used by the fibonacci action, including the
# goal message and the result message.
import actionlib_pkg.msg

# http://wiki.ros.org/actionlib_tutorials/Tutorials/Writing%20a%20Simple%20Action%20Client%20%28Python%29

def fibonacci_client():
    # Creates the SimpleActionClient, passing the type of the action
    # (FibonacciAction) to the constructor.
    client = actionlib.SimpleActionClient('fibonacci', actionlib_pkg.msg.fibonacciAction)

    # Waits until the action server has started up and started
    # listening for goals.
    client.wait_for_server()
    print("fibonacciActionClient.py - wait_for_server")
    goal = actionlib_pkg.msg.fibonacciGoal(order=20)  # Creates a goal to send to the action server.
    client.send_goal(goal)  # Sends the goal to the action server.
    # print("fibonacciActionClient.py - Call action")

    client.wait_for_result()  # Waits for the server to finish performing the action.
    return client.get_result()  # A FibonacciResult

if __name__ == '__main__':
    try:
        rospy.init_node('fibonacciActionClient')
        # print("fibonacciActionClient.py - Create node fibonacciActionClient")
        result = fibonacci_client()
        print("fibonacciActionClient.py -> Result:", ', '.join([str(n) for n in result.sequence]))
    except rospy.ROSInterruptException:
        print("fibonacciActionClient.py -> program interrupted before completion")